app/javascript/components/App.js

```diff
@@ -54,57 +53,65 @@ class App extends React.Component {
 
         <Switch>
           <Route exact path="/" component={Home} />
-
-          <Route path="/searchedit/:id" component={SearchEdit} />
-
-          {logged_in &&
+          {logged_in && (
             <Route
-              path="/mysearchindex"
+              path="/searchedit/:id"
               render={(props) => {
-                let user = current_user.id
-                let searches =
-                  this.state.searches.filter(search => search.user_id === user)
+                const id = props.match.params.id;
+                const search = this.state.searches.find(
+                  (search) => search.id === parseInt(id)
+                );
                 return (
-                  <MySearchIndex searches={searches}
+                  <SearchEdit
+                    updateSearch={this.updateSearch}
+                    current_user={current_user}
+                    search={search}
+                    logged_in={logged_in}
                   />
-                )
+                );
               }}
             />
-          }
-
+          )}
+          {logged_in && (
+            <Route
+              path="/mysearchindex"
+              render={(props) => {
+                let user = current_user.id;
+                let searches = this.state.searches.filter(
+                  (search) => search.user_id === user
+                );
+                return <MySearchIndex searches={searches} />;
+              }}
+            />
+          )}
           <Route
             path="/searchresults"
-            render={(props) =>
+            render={(props) => (
               <SearchResults
-                searches=
-                {this.state.searches.filter(search => search.id === 1)}
+                searches={this.state.searches.filter(
+                  (search) => search.id === 1
+                )}
+                logged_in={logged_in}
               />
-            }
+            )}
           />
-
           <Route
             path="/searchnew"
-            render={(props) =>
+            render={(props) => (
               <SearchNew
                 createNewSearch={this.createNewSearch}
                 current_user={current_user}
               />
-            }
+            )}
           />
-            }
-
+          }
           <Route
             path="/cityshow/:id"
             render={(props) => {
-              let cityName = props.match.params.id
-              let city = cityData.find(result => result.id === cityName)
-              return (
-                <CityShow
-                  city={city}
-                />
-              )
-            }
-            }
+              let cityName = props.match.params.id;
```

Comment:
nit-picky, but cityName feels like a bad name for this variable, perhaps 'currentCityId'

Reply:
done, changed name